extensions/functions_comparison.yaml

```diff
@@ -83,4 +83,36 @@ scalar_functions:
       - args:
           - value: fp64
         return: BOOLEAN
+  -
+    name: "is_finite"
+    description: Whether a value is finite.
+    impls:
+      - args:
+          - name: value_for_null
```

Comment:
What does `value_for_null` mean? Does it mean the value that should be returned from this function when a `NULL` is encountered?

Reply:
Yep, that's exactly it.  This PR is still a draft as I've not finished working out exactly what options might be needed with these functions.

Reply:
Interesting, I'm curious in what scenario the ability to return anything other than `NULL` for `is_(in)finite(NULL)` would be useful.

Reply:
There might not be any - currently looking at various implementations of this function to see what options are provided there, so I may end up concluding that it's not needed.

Reply:
Doesn't look like it's needed, removed now.